### PR TITLE
#29 Setup translation support in gulp build process

### LINKED ISSUE #29: Multi-language support
Setup the site to be able to translate into multiple languages:

- Design for language selector
- Integrate the solution - probably Transifex
- Workflow for translating and updating

### PR DESCRIPTION
**Pull request by laurentS**
This PR sets up translation for the website.

Each language comes out in its own folder, like:
- https://mapswipe.org/en/
- https://mapswipe.org/de/
etc...

## How it works
During gulp build/develop, the shtml files are compiled into html files. then each file is passed through the translation process, where each tag that has a `data-i18n` attribute will be translated, and the result inject into the output file. Whatever text was present before will be replaced, so you're free to write anything useful in there, or nothing at all.
The resulting translation is hardcoded in the output, so no need for any javascript on the client side to see the language.
This uses `i18next` to translate, so it's the same system as the app, so knowledge there should be usable here, mostly.

The gulp code is not very pretty, and not optimised, in part because it reloads everything for each file, but it all happens in less than 2 seconds, so no big deal.

## How to use it
This is only a functional skeleton. Each bit of text that needs translating must be marked with `data-i18n` attribute. Look at the example for `get involved` in the header file. There is no need to import any function or so since everything happens at build time.

All translations are stored in `locales/<lang>.json`, and follow the same format as the app.
For now, I've added `en` and `fr` as a starting point, look in `gulp-tasks/html.js` for the setup, hopefully I've commented enough that it is clear what is needed.

The root `/` is now empty, so you need to add a redirection to `/en/` for instance. I'll let you take care of that as an exercise :) If you deploy like this, the site will appear broken.

### CODE REVIEW

**Comment by jhenshall** [comment]
@laurentS you are a wizard! Thank you!! I will now have a look at the redirects, adding more text strings and integration with Transifex.

**Comment by jhenshall** [comment]
Looking good to me @laurentS @Hagellach37 but I hope you can confirm? 😬

**Review by jhenshall — APPROVED** [review]
Let's do this!

File: gulp-tasks/redirect.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+module.exports = function (gulp) {
+    const rename = require('gulp-rename');
```
Comment (laurentS) [inline gulp-tasks/redirect.js:6]: Maybe add a comment to explain what this does and how it works, so you don't need to work it out 12 months from now ;)

**Review by laurentS — COMMENTED** [review]
Just made a note about comments

**Pull request merged**